BDArmory/Guidances/MissileGuidance.cs

```diff
@@ -147,6 +147,184 @@ public static Vector3 GetAirToGroundTarget(Vector3 targetPosition, Vector3 targe
+            float termDist, ref int loftState, out float timeToImpact, out float targetDistance,
+            float minSpeed = 200)
+        {
+
+            Vector3 velDirection = missileVessel.Velocity().normalized;
+
+            targetDistance = Vector3.Distance(targetPosition, missileVessel.transform.position);
+
+            float currSpeed = Mathf.Max((float)missileVessel.srfSpeed, minSpeed);
+            Vector3 currVel = currSpeed * velDirection;
+
+            //Vector3 Rdir = (targetPosition - missileVessel.transform.position).normalized;
+            //float rDot = Vector3.Dot(targetVelocity - currVel, Rdir);
+
+            float leadTime = (float)(1 / ((targetVelocity - currVel).magnitude / targetDistance));
+            //float leadTime = (targetDistance / rDot);
+
+            timeToImpact = leadTime;
+            leadTime = Mathf.Clamp(leadTime, 0f, 8f);
+
+            // If loft is not terminal
+            if ((targetDistance > termDist) && (loftState < 3))
+            {
+                if (BDArmorySettings.DEBUG_MISSILES) Debug.Log("[BDArmory.MissileLauncher]: Lofting");
+
+                // Get up direction
+                Vector3 upDirection = VectorUtils.GetUpDirection(missileVessel.CoM);
+
+                // Use the gun aim-assist logic to determine ballistic angle (assuming no drag)
+                Vector3 bulletRelativePosition, bulletRelativeVelocity, bulletAcceleration, bulletRelativeAcceleration, targetPredictedPosition, bulletDropOffset, lastVelDirection, ballisticTarget, targetHorVel, targetCompVel;
+                
+                var firePosition = missileVessel.transform.position + (currSpeed * velDirection) * Time.fixedDeltaTime; // Bullets are initially placed up to 1 frame ahead (iTime). Not offsetting by part vel gives the correct initial placement.
+                bulletRelativePosition = targetPosition - firePosition;
+                float timeToCPA = timeToImpact; // Rough initial estimate.
+                targetPredictedPosition = AIUtils.PredictPosition(targetPosition, targetVelocity, targetAcceleration, timeToCPA);
+
+                // Velocity Compensation Logic
+                float compMult = Mathf.Clamp(0.5f * (targetDistance - termDist) / termDist, 0f, 1f);
+                Vector3 velDirectionHor = (velDirection - upDirection * Vector3.Dot(velDirection, upDirection)).normalized;
+                targetHorVel = targetVelocity - upDirection * Vector3.Dot(targetVelocity, upDirection); // Get target horizontal velocity (relative to missile frame)
+                float targetAlVelMag = Vector3.Dot(targetHorVel, velDirectionHor); // Get magnitude of velocity aligned with the missile velocity vector (in the horizontal axis)
+                targetAlVelMag *= Mathf.Sign(velComp) * compMult;
+                targetAlVelMag = 0.5f * (targetAlVelMag + Mathf.Abs(targetAlVelMag)); // Set -ve velocity (I.E. towards the missile) to 0 if velComp is +ve, otherwise for -ve
```

Comment:
This is just `targetAlVelMag = Mathf.Max(targetAlVelMag, 0);`.

Reply:
Fixed